src/index.ts

```diff
@@ -428,29 +428,26 @@ class Envoy {
       // ZomeCall to Conductor App Interface
       let zomeCall_response, holo_error
       try {
-        const zomeCallArgs = (typeof call_spec.args === 'object')
-          ? Object.entries(call_spec.args).map(([k, v]) => {
-            if (!k || !v) return {};
-            return `${k} : ${typeof v}`
-          }).join(", ")
-          : call_spec.args
-        log.debug("Calling zome function %s->%s( %s ) on cell_id (%s), cap token (%s), and provenance (%s):", () => [
-          call_spec.zome, call_spec.function, zomeCallArgs, call_spec.cell_id, null, agent_id]);
-
-        // In case of no call args, convert empty obj to null
-        if (Object.keys(call_spec.args).length <= 0) {
-          log.debug('No call_spec.args, converting value to null for zomeCall.');
-          call_spec.args = null
-        };
-
         const hosted_app_cell_id = call_spec["cell_id"];
+        let payload = call_spec["args"]
+        log.silly('Original payload for zomeCall:', payload);
+        if (typeof payload === 'string') {
+          payload = msgpack.decode(Buffer.from(call_spec["args"], 'base64'));
```

Comment:
I would prefer to use a new variable rather than changing the type of an existing variable (from `string` to `any`)